### PR TITLE
Filter out empty array parameters to stop openshift errors.

### PR DESCRIPTION
**Pull request by singularo**
I was removing things from various parts of the shepherd code, but doing it in the client makes more sense or it will be like whack a mole.

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -328,12 +328,29 @@ public function getGuzzleClient() {
     return $this->guzzleClient;
   }
 
+  public function filterEmptyArrays($array) {
```
Comment (fenetikm) [inline src/Client.php]: missing doc block

Comment (fenetikm) [reply src/Client.php]: this would be a great candidate for some tests so that one could see what some of the inputs this is expected to handle.

Comment (fenetikm) [reply src/Client.php]: should it be `filterEmptyArrays(array $array)` ?

Comment (pingers) [reply src/Client.php]: Can't be type array because recursive :(

Comment (pingers) [reply src/Client.php]: This method shouldn't be public - it's not part of the purpose of this class - just internal functionality that other classes shouldn't rely on.

Comment (singularo) [reply src/Client.php]: Actually, it will always be an array, so yeah, can do that, and docblock added.

**Comment by singularo** [comment]
This fixes the remaining error for 3.11 support @pingers

**Review by pingers — APPROVED** [review]
:shipit:

**Pull request merged**